frmts/ers/ersdataset.cpp

```diff
@@ -828,11 +828,18 @@ int ERSDataset::Identify( GDALOpenInfo * poOpenInfo )
 /* -------------------------------------------------------------------- */
 /*      We assume the user selects the .ers file.                       */
 /* -------------------------------------------------------------------- */
-    if( poOpenInfo->nHeaderBytes < 15
-        || !STARTS_WITH_CI((const char *) poOpenInfo->pabyHeader, "DatasetHeader ") )
-        return FALSE;
+    if( poOpenInfo->nHeaderBytes >= 15
+        && STARTS_WITH_CI((const char *) poOpenInfo->pabyHeader, "DatasetHeader ") )
+        return TRUE;
+
+/* -------------------------------------------------------------------- */
+/*      We assume the user selects the .ers file.                       */
+/* -------------------------------------------------------------------- */
+    if( poOpenInfo->nHeaderBytes >= 1
+        && STARTS_WITH_CI((const char *) poOpenInfo->pabyHeader, "#") )
+        return TRUE;
```

Comment:
We cannot just identify a ERS dataset if its first character is #. This is not specific enough . I'd suggest just to look for the presence of "DatasetHeader " anywhere in the starting bytes of the file:

```suggestion
    if( poOpenInfo->nHeaderBytes >= 15
        && strstr((const char *) poOpenInfo->pabyHeader, "DatasetHeader ") != nullptr )
        return TRUE;
```

Reply:
Stand by for a tweak to deal with the `data/ers/caseinsensitive.ers` file...

Reply:
When I re-ran the tests before, I re-ran them without _re-compiling_ -- have been dealing with too much Python lately.  Sorry about that.  

I modified the file type detection to use `CPLString`.  Since we need to a case-insensitive search of the header for the `DatasetHeader ` string, neither the `STARTS_WITH_CI` macro nor `strstr` are powerful enough.  Instantiating a `CPLString` allows us to easily use the `ifind` method, which is powerful enough for this search.